sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -0,0 +1,506 @@
+            disbursementSchedule: loan.relatedSchedule as DisbursementSchedule,
+            disbursementValueCode: valueCode,
+            overawardValue: (
+              +loan.awardValue.overawardAmountSubtracted * -1
+            ).toString(),
+            originType: DisbursementOverawardOriginType.AwardValueAdjusted,
+            creator: auditUser,
+          } as DisbursementOveraward);
+        }
+      }
+    }
+  }
+
+  /**
+   * Determine when a BC Full-time funding should not be disbursed.
+   * In this case the e-Cert can still be generated with th federal funding.
+   * @param schedule disbursement to be checked.
+   * @param disbursementValue award to be checked.
+   * @returns true if the funding should not be disbursed, otherwise, false.
+   */
+  private shouldStopFunding(
+    schedule: ECertDisbursementSchedule,
+    disbursementValue: DisbursementValue,
+  ): boolean {
+    return (
+      schedule.stopFullTimeBCFunding &&
+      schedule.studentAssessment.application.currentAssessment.offering
+        .offeringIntensity === OfferingIntensity.fullTime &&
+      BC_FUNDING_TYPES.includes(disbursementValue.valueType)
+    );
+  }
+
+  /**
+   * Get the list of the awards from all disbursements with the associated
+   * schedule. Useful to generate the overaward balance deduction record.
+   * @param disbursementSchedules schedules to have the awards listed.
+   * @param valueTypes types to be part of the final awards list.
+   * @returns awards filtered by the types requested with the associated
+   * disbursement schedule.
+   */
+  private getAwardsByTypeWithRelatedSchedule(
+    disbursementSchedules: ECertDisbursementSchedule[],
+    valueTypes: DisbursementValueType[],
+  ): AwardValueWithRelatedSchedule[] {
+    const result: AwardValueWithRelatedSchedule[] = [];
+    for (const relatedSchedule of disbursementSchedules) {
+      for (const awardValue of relatedSchedule.disbursementValues) {
+        if (valueTypes.includes(awardValue.valueType)) {
+          result.push({ relatedSchedule, awardValue });
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Try to deduct an overaward balance owed by the student due to some previous overaward in some
+   * previous application.
+   * @param awards specific loan award being adjusted (e.g CSLF, BCSL). This will contain one or
+   * two entries, always from the same award, from where the student debit will be deducted.
+   * The debit will try to be settle as much as possible with the first award. If it is not enough
+   * if will check for the second award (second disbursement), when present.
+   * The awards lists will be always from the same loan award code. For instance, the list list
+   * will contain one or two awards of type CSLF.
+   * @param overawardBalance total overaward balance be deducted.
+   */
+  private subtractOverawardBalance(
+    awards: DisbursementValue[],
+    overawardBalance: number,
+  ) {
```

Comment:
return type?

Reply:
The method is not returning anything, are you asking to add a `:void` as the return type?

Reply:
https://github.com/bcgov/SIMS/pull/1644/files/65ab00fa082429ca6a79e88ee08c6f7396c09efd#r1066449140

Reply:
@guru-aot @andrepestana-aot @dheepak-aot can we start/continue to add the explicit `:void` for methods that do not have the return types?
Please keep in mind Promise<void> is different than `void`. The point here is just about the `void` itself.

Reply:
I think it's good to be explicit about the return type even if it is void or Promise<void>.